src/dex/uniswap-v3/uniswap-v3.ts

```diff
@@ -0,0 +1,630 @@
+      await Promise.all(
+        this.supportedFees.map(async fee =>
+          this.getPool(_srcAddress, _destAddress, fee, blockNumber),
+        ),
+      )
+    ).filter(pool => pool);
+
+    if (pools.length === 0) return [];
+
+    return pools.map(pool =>
+      this.getPoolIdentifier(_srcAddress, _destAddress, pool!.feeCode),
+    );
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<UniswapV3Data>> {
+    try {
+      const _srcToken = this.dexHelper.config.wrapETH(srcToken);
+      const _destToken = this.dexHelper.config.wrapETH(destToken);
+
+      const [_srcAddress, _destAddress] = this._getLoweredAddresses(
+        _srcToken,
+        _destToken,
+      );
+
+      if (_srcAddress === _destAddress) return null;
+
+      let selectedPools: UniswapV3EventPool[] = [];
+      if (limitPools === undefined) {
+        for (const fee of this.supportedFees) {
+          let pool =
+            this.eventPools[
+              this.getPoolIdentifier(_srcAddress, _destAddress, fee)
+            ];
+          if (!pool) {
+            pool = await this.getPool(
+              _srcAddress,
+              _destAddress,
+              fee,
+              blockNumber,
+            );
+          }
+          if (pool) {
+            selectedPools.push(pool);
+          }
+        }
+      } else {
+        const pairIdentifierWithoutFee = this.getPoolIdentifier(
+          _srcAddress,
+          _destAddress,
+          0n,
+          // Trim from 0 fee postfix, so it become comparable
+        ).slice(0, -1);
+
+        const poolIdentifiers = limitPools.filter(identifier =>
+          identifier.startsWith(pairIdentifierWithoutFee),
+        );
+        for (const identifier of poolIdentifiers) {
+          let pool = this.eventPools[identifier];
+          if (!pool) {
+            const [, srcAddress, destAddress, fee] = identifier.split('_');
+            pool = await this.getPool(
+              srcAddress,
+              destAddress,
+              BigInt(fee),
+              blockNumber,
+            );
+          }
+
+          if (pool) {
+            selectedPools.push(pool);
+          }
+        }
+      }
+
+      if (selectedPools.length === 0) return null;
+
+      const states = selectedPools.map(pool => {
+        let state = pool.getState(blockNumber);
+        // if (state === null || !state.isValid) {
+        //   if (state === null) {
+        //     this.logger.trace(
+        //       `${this.dexKey}: State === null. Generating new one`,
+        //     );
+        //   } else if (!state.isValid) {
+        //     this.logger.trace(
+        //       `${this.dexKey}: State is invalid. Generating new one`,
+        //     );
+        //   }
+
+        // state = await pool.generateState(blockNumber);
+        // pool.setState(state, blockNumber);
+        // }
```

Comment:
Maybe better to remove this logic if it is not used anymore

Reply:
Let me check that is not needed